zrml/court/src/migrations.rs

```diff
@@ -14,3 +14,220 @@
+
+const COURT_REQUIRED_STORAGE_VERSION: u16 = 2;
+const COURT_NEXT_STORAGE_VERSION: u16 = 3;
+
+pub struct MigrateCourtPoolItems<T>(PhantomData<T>);
+
+impl<T> OnRuntimeUpgrade for MigrateCourtPoolItems<T>
+where
+    T: Config,
+{
+    fn on_runtime_upgrade() -> Weight {
+        let mut total_weight = T::DbWeight::get().reads(1);
+        let market_commons_version = StorageVersion::get::<Court<T>>();
+        if market_commons_version != COURT_REQUIRED_STORAGE_VERSION {
+            log::info!(
+                "MigrateCourtPoolItems: court storage version is {:?}, but {:?} is required",
+                market_commons_version,
+                COURT_REQUIRED_STORAGE_VERSION,
+            );
+            return total_weight;
+        }
+        log::info!("MigrateCourtPoolItems: Starting...");
+
+        let res = crate::CourtPool::<T>::translate::<OldCourtPoolOf<T>, _>(|old_pool_opt| {
+            old_pool_opt.map(|mut old_pool| {
+                <CourtPoolOf<T>>::truncate_from(
+                    old_pool
+                        .iter_mut()
+                        .map(|old_pool_item: &mut OldCourtPoolItemOf<T>| CourtPoolItemOf::<T> {
+                            stake: old_pool_item.stake,
+                            court_participant: old_pool_item.court_participant.clone(),
+                            consumed_stake: old_pool_item.consumed_stake,
+                            joined_at: old_pool_item.joined_at,
+                            uneligible_index: old_pool_item
+                                .joined_at
+                                .checked_div(&T::InflationPeriod::get())
+                                // because inflation period is not zero checked_div is safe
+                                // however, if not use the current period index
+                                // 5_184_000 is the block of the inflation reward distribution
+                                // 5_184_000 / 216_000 (blocks per 30 day inflation period) = 24
+                                .unwrap_or(24u64.saturated_into::<BlockNumberOf<T>>()),
+                            // using old_pool_item.stake leads to all joins in period 24
+                            // to be uneligible, which is exactly what we want
+                            // if using zero, all joins of period 24 would be eligible,
+                            // but only joins in 23 waited the full period yet
+                            // to understand the calculation of the eligible stake look into handle_inflation
+                            uneligible_stake: old_pool_item.stake,
+                        })
+                        .collect::<Vec<_>>(),
+                )
+            })
+        });
+        match res {
+            Ok(_) => log::info!("MigrateCourtPoolItems: Success!"),
+            Err(e) => log::error!("MigrateCourtPoolItems: Error: {:?}", e),
+        }
+
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
```

Comment:
One more read for fetching the old court pool vector.

```suggestion
        total_weight = total_weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
```

Reply:
9067fbc17dffd7bd1952b3d76e689e466b5ab515